src/main/java/com/amazon/opendistroforelasticsearch/sql/query/AggregationQueryAction.java

```diff
@@ -231,10 +239,34 @@ private AggregationBuilder getGroupAgg(Field field, Select select2) throws SqlPa
         }
 
         if (!refrence) lastAgg = aggMaker.makeGroupAgg(field);
-        
+
+        // find if we have order for that aggregation. As of now only special case for script fields
+        if (field.isScriptField()) {
+            addOrderByScriptFieldIfPresent(select, lastAgg, field.getExpression());
+        }
+
         return lastAgg;
     }
 
+    private void addOrderByScriptFieldIfPresent(Select select, AggregationBuilder groupByAggregation, SQLExpr groupByExpression) {
+        // TODO: Explore other ways to correlate different fields/functions in the query (params?)
+        // This feels like a hacky way, but it's the best that could be done now.
+
+        Optional<Order> orderForGroupBy =
+                select
+                    .getOrderBys()
+                    .stream()
+                    .filter(order -> groupByExpression.equals(order.getSortField().getExpression()))
+                    .findFirst();
+
+        if (!orderForGroupBy.isPresent()) {
+            return;
+        }
+
+        TermsAggregationBuilder termsAggregationBuilder = (TermsAggregationBuilder) groupByAggregation;
```

Comment:
Just an option, you may be able to put this logic into  `ifPresent(Consumer)` which could append after `.findFirst()` in chain. And the `if` check above could be removed. Then the code would be exactly consistent with your method name.

```
select...
      .findFirst() 
      .ifPresent(...);
```

Reply:
Good idea, thanks. Done